### DISCUSSION TITLE
[SDK] Issue finalizing withdrawal on L1

### DISCUSSION THREAD

**Discussion by tradable-kash** [zkSync-Community-Hub/zksync-developers#83] [SDKs] [answered]
### Environment

Testnet

### Acknowledgement

- [X] I understand that issues are expected with the Python and Java SDK as they are being updated.

### Issue Description

Hello,

While I read the disclaimer above regarding zksync2 Python SDK being WIP, wanted to still get this in front of you guys. While trying my hands out on examples 09_withdrawal.py, I had success but after 24 hrs when I tried to run 10_finalize_withdrawal.py - I did not have a single successful run. I started seeing the following unhelpful error message [jj means nothing to me] -

  File "/.pyenv/versions/3.9.0/lib/python3.9/site-packages/web3/manager.py", line 73, in apply_error_formatters
    formatted_resp = pipe(response, error_formatters)
  File "cytoolz/functoolz.pyx", line 680, in cytoolz.functoolz.pipe
  File "cytoolz/functoolz.pyx", line 655, in cytoolz.functoolz.c_pipe
  File ".pyenv/versions/3.9.0/lib/python3.9/site-packages/web3/_utils/method_formatters.py", line 685, in raise_contract_logic_error_on_revert
    raise ContractLogicError(response["error"]["message"])
web3.exceptions.ContractLogicError: execution reverted: jj

Please check the successful [Withdraw to L1 on Testnet](https://goerli.explorer.zksync.io/tx/0x848ab1f48837c64f44c142a70b19664b1f9f9db10a23a018610ba1056a51ca30) 

### Expected Behavior

I expected 2 things - 
1) Helpful error message to debug on my own rather reach you for this trivial issue [I think]
2) Expected examples to work flawless after replacing ENV VARS and thus finalizing Withdraw on L1

### Code Example

https://github.com/zksync-sdk/zksync2-python/blob/master/examples/10_finalize_withdrawal.py

**Comment by dutterbutter** [comment]
Hello 👋 

I'm sorry for the inconvenience you've experienced 😔. This is a frustrating error message to see. We definitely need to improve our error messaging in such cases 🛠️. I'll investigate the issue with the example you've mentioned and work on correcting it. Thank you for bringing this to our attention and for your patience as we continue to refine our SDKs 🙏.

**Reply by dutterbutter** [reply]
I was able to reproduce so we will address this issue ASAP! 🚀

**Comment by danijelTxFusion** [comment, marked as answer]
The problem was that withdrawal transactions are [automatically finalized](https://era.zksync.io/docs/reference/concepts/bridging-asset.html#withdrawals-to-l1) on testnet. That what [jj](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/facets/Mailbox.sol#L176) error code stands for. Therefore, on the testnet, there is no need to execute `FinalizeWithdraw`. However, you are correct when it comes to error handling; it should be more helpful and user-friendly.

**Comment by tradable-kash** [comment]
Thank you for the clarity. I see the Note as well on site.
We can close

On Tue, Aug 8, 2023 at 8:17 AM Danijel Radakovic ***@***.***>
wrote:

> The problem was that withdrawal transactions are automatically finalized
> <https://era.zksync.io/docs/reference/concepts/bridging-asset.html#withdrawals-to-l1>
> on testnet. That what jj
> <https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/facets/Mailbox.sol#L176>
> error code stands for. Therefore, on the testnet, there is no need to
> execute FinalizeWithdraw. However, you are correct when it comes to error
> handling; it should be more helpful and user-friendly.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zkSync-Community-Hub/zkync-developers/discussions/83#discussioncomment-6668783>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OKCZG5UQW32DFEGBM2OV3XUIU6RANCNFSM6AAAAAA3BXLTYM>
> .
> You are receiving this because you authored the thread.Message ID:
> <zkSync-Community-Hub/zkync-developers/repo-discussions/83/comments/6668783
> @github.com>
>

-- 
This email and any files transmitted with it are confidential, private and 
intended solely for the use of the individual or entity to whom they are 
addressed. If you are not the intended recipient you are notified that 
disclosing, copying, distributing or taking any action in reliance on the 
contents of this information is strictly prohibited.
